Applications/AlexNet/jni/cifar_dataloader.h

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   cifar_dataloader.h
+ * @date   24 Jun 2021
+ * @brief  dataloader for cifar 100
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+#include <tensor_dim.h>
+
+#include <fstream>
+#include <random>
+#include <string>
+#include <vector>
+
+namespace nntrainer::alex {
+
+using TensorDim = ml::train::TensorDim;
+
+/**
+ * @brief DataLoader interface used to load cifar data
+ *
+ */
+class DataLoader {
+public:
+  /**
+   * @brief Destroy the Data Loader object
+   *
+   */
+  virtual ~DataLoader() {}
+
+  /**
+   * @brief create an iteration to fed to the generator callback
+   *
+   * @param[out] input list of inputs that is already allocated by nntrainer,
+   * and this function is obliged to fill
+   * @param[out] label list of label that is already allocated by nntrainer, and
+   * this function is obliged to fill
+   * @param[out] last  optional property to set when the epoch has finished
+   */
+  virtual void next(float **input, float **label, bool *last) = 0;
+};
+
+/**
+ * @brief RandomData Generator
+ *
+ */
+class RandomDataLoader final : public DataLoader {
+public:
+  /**
+   * @brief Construct a new Random Data Loader object
+   *
+   * @param input_shapes input_shapes with appropriate batch
+   * @param output_shapes label_shapes with appropriate batch
+   * @param iteration     iteration per epoch
+   */
+  RandomDataLoader(const std::vector<TensorDim> &input_shapes,
+                   const std::vector<TensorDim> &output_shapes, int iteration);
+
+  /**
+   * @brief Destroy the Random Data Loader object
+   *
+   */
+  ~RandomDataLoader() {}
+
+  /**
+   * @copydoc void DataLoader::next(float **input, float**label, bool *last)
+   */
+  void next(float **input, float **label, bool *last);
+
+private:
+  unsigned int iteration;
+  unsigned int iteration_for_one_epoch;
+
+  std::vector<TensorDim> input_shapes;
+  std::vector<TensorDim> output_shapes;
+
+  std::mt19937 rng;
```

Comment:
`RandomDataLoader` and `Cifar100DataLoader` classes have its own `rng` object as below.


However, Both of them have the same parent `DataLoader` class. So what about moving 'rng' object to `DataLoader` class?

Reply:
moving 'rng' to ```DataLoader```class and conduct Run Test (There is no Problem)
``` cpp
class DataLoader {
public:
  /**
   * @brief Destroy the Data Loader object
   */
  virtual ~DataLoader() {}

  /**
   * @brief create an iteration to fed to the generator callback
   *
   * @param[out] input list of inputs that is already allocated by nntrainer,
   * and this function is obliged to fill
   * @param[out] label list of label that is already allocated by nntrainer, and
   * this function is obliged to fill
   * @param[out] last  optional property to set when the epoch has finished
   */
  virtual void next(float **input, float **label, bool *last) = 0;

protected:
  std::mt19937 rng;
};
```

Reply:
Good job! :+1: